.circleci/config.yml

```diff
@@ -0,0 +1,67 @@
+version: 2.1
+
+defaults: &defaults
+  working_directory: &working_directory ~/mavenlink/mavenlink_gem/current
+  parallelism: 1
+  shell: /bin/bash --login
+  environment:
+    CIRCLE_ARTIFACTS: /tmp/circleci-artifacts
+    CIRCLE_TEST_REPORTS: /tmp/circleci-test-results
+    RACK_ENV: "test"
+    RAILS_ENV: "test"
+  docker:
+  - image: circleci/ruby:2.2-jessie-node
+    command: /sbin/init
+
+workflows:
+  version: 2
+  test:
+    jobs:
+      - build
+      - rspec:
+          requires:
+            - build
+
+jobs:
+  build:
+    <<: *defaults
+    steps:
+    - checkout
+    - run: mkdir -p $CIRCLE_ARTIFACTS $CIRCLE_TEST_REPORTS
+    - restore_cache:
+        keys:
+        - v1-dep-{{ .Branch }}-
+        - v1-dep-master-
+        - v1-dep-
+    - run: gem install bundler -v "=1.13.6"
+    - run: mkdir -p tmp/cache
+    - run:
+        name: bundle
+        command: |
+          bundle check --path=~/mavenlink/mavenlink_gem/bundle || bundle install --path=~/mavenlink/mavenlink_gem/bundle --jobs=4 --retry=3
```

Comment:
I _think_ theres a mismatch here with `mavenlink/mavenlink_gem/bundle` vs `mavenlinkn/mavenlink_gem/current/bundle` thats causing your gem persistence issue. I'd consider removing the idea of `current` from your working directory default and persist to workspace bits to resolve it.